File: testdata/programs/bench_fib.py
import time

def fib(n: int) -> int:
    if n <= 1:
        return n
    return fib(n - 1) + fib(n - 2)

started = time.perf_counter()
result = fib(35)
ended = time.perf_counter()

print("python recursion benchmark")
print(f"fib(35)={result}")
print(f"elapsed_ms={(ended - started) * 1000:.3f}")

File: testdata/programs/bench_fib.lua
local function fib(n)
  if n <= 1 then return n end
  return fib(n - 1) + fib(n - 2)
end
local t = os.clock()
local r = fib(35)
local e = os.clock()
print("gopher-lua recursion benchmark")
print("fib(35)=" .. r)
print("elapsed_ms=" .. ((e - t) * 1000.0))
